doc/manuals/orion-api.md

```diff
@@ -2229,22 +2209,136 @@ Let's consider the following `notification.httpCustom` object in a given subscri
+    exactly with the macro expression. Thus, `"t": "${temperature}"` works, but
+    `"t": "the temperature is ${temperature}"` or `"h": "humidity ranges from ${humidityMin} to ${humidityMax}"`
+    will not work
+  * It takes into account the nature of the attribute value to be replaced. For instance,
+    `"t": "${temperature}"` resolves to `"t": 10` if temperature attribute is a number or to
+    `"t": "10"` if `temperature` attribute is a string.
+  * If the attribute doesn't exist in the entity, then `null` value is used
+* `Content-Type` header is set to `application/json`, except if overwritten by `headers` field
+
+### NGSI payload patching
+
+If `ngsi` is used in `httpCustom` or `mqttCustom` the following considerations apply.
+Note that only one of the following can be used a the same time: `payload`, `json` or `ngsi.
+
+The `ngsi` field can be used to specify an entity fragment that will *patch* the entity in
+the notification. This allows to add new attributes and/or change the value of
+existing attributes, id and type. The resulting notification uses NGSIv2 normalized format described
+in [Notification Messages](#notification-messages).
+
+For instance:
+
+```
+"httpCustom": {
+   ...
+   "ngsi": {
+     "id": "prefix:${id}",
+     "type": "newType",
+     "originalService": {
+       "value": "${service}",
+       "type": "Text"
+     },
+     "originalServicePath": {
+       "value": "${servicePath}",
+       "type": "Text"
+     }
+   }
+}
+```
+
+Some notes to take into account when using `ngsi`:
+
+* The value of the `ngsi` field must be a valid [JSON Entity Representation](#json-entity-representation),
+  with some extra considerations:
+  * `id` or `type` are not mandatory
+  * If attribute `type` is not specified, the defaults described in [Partial Representations](#partial-representations)
+    are used.
+  * Attribute `metadata` is not allowed
+  * `{}` is a valid value for the `ngsi` field, in which case no patching is done and the original
+    notification is sent
+* If `notification.attrs` is used, the attribute filtering is done *after* applyig the NGSI patching
+* The patching semantic applied is *update or append* (similar to `append` `actionType` in updates) but other
+  semantics could be added in the future.
+* The [macro replacement logic](#macro-substitution) works as expected, with the following
+  considerations:
+  * It cannot be used in the key part of JSON objects, i.e. `"${key}": 10` will not work
+  * It cannot be used in the attribute `type`. Only in the `value` macro replacements can be done.
+  * If the macro *covers completely the string where is used*, then the JSON nature of the attribute value
+    is taken into account. For instance, `"value": "${temperature}"` resolves to `"value": 10`
+    if temperature attribute is a number or to `"value": "10"` if `temperature` attribute is a string.
+  * If the macro *is only part of string where is used*, then the attributue value is always casted
```

Comment:
```suggestion
  * If the macro *is only part of string where is used*, then the attribute value is always casted
```

Reply:
Fixed in commit [218a7d7](https://github.com/telefonicaid/fiware-orion/pull/4229/commits/218a7d7cbd694ee1a35a9f417cfdacc41bbcc263)